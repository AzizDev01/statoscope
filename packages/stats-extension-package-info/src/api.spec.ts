import APIFactory from './api';
import Generator from './generator';

test('should work', () => {
  const generator = new Generator();

  generator.handleInstance('foo-compilation', 'foo-package', 'foo-instance', {
    version: '1.0.0',
  });
  generator.handleInstance('foo-compilation', 'foo-package', 'bar-instance', {
    version: '2.0.0',
  });
  generator.handleInstance('foo-compilation', 'bar-package', 'bar-instance', {
    version: '3.0.0',
  });
  generator.handleInstance('bar-compilation', 'foo-package', 'bar-instance', {
    version: '4.0.0',
  });

  const data = generator.get();
  const api = APIFactory(data);

  expect(api.getPackage('foo', 'bar')).toBeNull();
  expect(api.getPackage('foo-compilation', 'bar')).toBeNull();
  expect(api.getPackage('foo-compilation', 'foo-package')).toMatchInlineSnapshot(`
    Object {
      "instances": Array [
        Object {
          "info": Object {
            "version": "1.0.0",
          },
          "path": "foo-instance",
        },
        Object {
          "info": Object {
            "version": "2.0.0",
          },
          "path": "bar-instance",
        },
      ],
      "name": "foo-package",
    }
  `);
  expect(api.getPackage('foo-compilation', 'bar-package')).toMatchInlineSnapshot(`
    Object {
      "instances": Array [
        Object {
          "info": Object {
            "version": "3.0.0",
          },
          "path": "bar-instance",
        },
      ],
      "name": "bar-package",
    }
  `);
  expect(api.getPackage('bar-compilation', 'foo-package')).toMatchInlineSnapshot(`
    Object {
      "instances": Array [
        Object {
          "info": Object {
            "version": "4.0.0",
          },
          "path": "bar-instance",
        },
      ],
      "name": "foo-package",
    }
  `);

  expect(api.getInstance('foo', 'bar', 'bar')).toBeNull();
  expect(api.getInstance('foo-compilation', 'bar', 'baz')).toBeNull();
  expect(api.getInstance('foo-compilation', 'foo-package', 'bar')).toBeNull();
  expect(api.getInstance('foo-compilation', 'foo-package', 'foo-instance'))
    .toMatchInlineSnapshot(`
    Object {
      "info": Object {
        "version": "1.0.0",
      },
      "path": "foo-instance",
    }
  `);
  expect(api.getInstance('foo-compilation', 'foo-package', 'bar-instance'))
    .toMatchInlineSnapshot(`
    Object {
      "info": Object {
        "version": "2.0.0",
      },
      "path": "bar-instance",
    }
  `);
  expect(api.getInstance('foo-compilation', 'bar-package', 'bar-instance'))
    .toMatchInlineSnapshot(`
    Object {
      "info": Object {
        "version": "3.0.0",
      },
      "path": "bar-instance",
    }
  `);
  expect(api.getInstance('bar-compilation', 'foo-package', 'bar-instance'))
    .toMatchInlineSnapshot(`
    Object {
      "info": Object {
        "version": "4.0.0",
      },
      "path": "bar-instance",
    }
  `);
});
