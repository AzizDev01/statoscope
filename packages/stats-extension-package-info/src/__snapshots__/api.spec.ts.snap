// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should work 1`] = `
Object {
  "instances": Array [
    Object {
      "info": Object {
        "version": "1.0.0",
      },
      "path": "foo-instance",
    },
    Object {
      "info": Object {
        "version": "2.0.0",
      },
      "path": "bar-instance",
    },
  ],
  "name": "foo-package",
}
`;

exports[`should work 2`] = `
Object {
  "instances": Array [
    Object {
      "info": Object {
        "version": "3.0.0",
      },
      "path": "bar-instance",
    },
  ],
  "name": "bar-package",
}
`;

exports[`should work 3`] = `
Object {
  "instances": Array [
    Object {
      "info": Object {
        "version": "4.0.0",
      },
      "path": "bar-instance",
    },
  ],
  "name": "foo-package",
}
`;

exports[`should work 4`] = `
Object {
  "info": Object {
    "version": "1.0.0",
  },
  "path": "foo-instance",
}
`;

exports[`should work 5`] = `
Object {
  "info": Object {
    "version": "2.0.0",
  },
  "path": "bar-instance",
}
`;

exports[`should work 6`] = `
Object {
  "info": Object {
    "version": "3.0.0",
  },
  "path": "bar-instance",
}
`;

exports[`should work 7`] = `
Object {
  "info": Object {
    "version": "4.0.0",
  },
  "path": "bar-instance",
}
`;
